homeassistant/components/yale_smart_alarm/coordinator.py

```diff
@@ -110,9 +128,17 @@ def get_updates(self) -> dict:
         """Fetch data from Yale."""
 
         if self.yale is None:
-            self.yale = YaleSmartAlarmClient(
-                self.entry.data[CONF_USERNAME], self.entry.data[CONF_PASSWORD]
-            )
+            try:
+                self.yale = YaleSmartAlarmClient(
+                    self.entry.data[CONF_USERNAME], self.entry.data[CONF_PASSWORD]
+                )
+            except AuthenticationError as error:
+                self.auth_failed(error)
+                raise UpdateFailed from error
```

Comment:
You probably want `raise ConfigEntryAuthFailed` because that starts a reauth for you.  `UpdateFailed` seems be for more of a temporary problem, not involving authentication.